### PR TITLE
Nullability annotations for System.Security.Cryptography.Encoding

### PR DESCRIPTION
**Pull request by manne**

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs [hunk 1]
```diff
@@ -21,12 +21,12 @@ internal static partial class OidLookup
         //
         // Attempts to map a friendly name to an OID. Returns null if not a known name.
         //
-        public static string ToFriendlyName(string oid, OidGroup oidGroup, bool fallBackToAllGroups)
+        public static string? ToFriendlyName(string oid, OidGroup oidGroup, bool fallBackToAllGroups)
         {
-            if (oid == null)
+            if (oid is null)
```
Comment (danmosemsft) [inline src/libraries/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs]: Why change == to is?

File: src/libraries/Common/src/Interop/Windows/BCrypt/Cng.cs [hunk 2]
```diff
@@ -70,8 +69,7 @@ public enum OpenAlgorithmProviderFlags : int
 
         public static SafeAlgorithmHandle BCryptOpenAlgorithmProvider(string pszAlgId, string pszImplementation, OpenAlgorithmProviderFlags dwFlags)
         {
-            SafeAlgorithmHandle hAlgorithm = null;
-            NTSTATUS ntStatus = Interop.BCryptOpenAlgorithmProvider(out hAlgorithm, pszAlgId, pszImplementation, (int)dwFlags);
+            NTSTATUS ntStatus = Interop.BCryptOpenAlgorithmProvider(out SafeAlgorithmHandle hAlgorithm, pszAlgId, pszImplementation, (int)dwFlags);
```
Comment (bartonjs) [inline src/libraries/Common/src/Interop/Windows/BCrypt/Cng.cs]: Please undo this style change

File: src/libraries/Common/src/Interop/Windows/BCrypt/Cng.cs [hunk 3]
```diff
@@ -243,11 +241,11 @@ protected sealed override bool ReleaseHandle()
 
     internal sealed class SafeKeyHandle : SafeBCryptHandle
     {
-        private SafeAlgorithmHandle _parentHandle = null;
+        private SafeAlgorithmHandle? _parentHandle = null;
 
         public void SetParentHandle(SafeAlgorithmHandle parentHandle)
         {
-            Debug.Assert(_parentHandle == null);
+            Debug.Assert(_parentHandle is null);
```
Comment (bartonjs) [inline src/libraries/Common/src/Interop/Windows/BCrypt/Cng.cs]: As SafeHandle doesn't define operator== there's no difference between `is null` and `== null` on this line.

Please don't make style changes in this PR.

File: src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedData.cs [hunk 4]
```diff
@@ -46,7 +53,7 @@ public Oid Oid
 
             set
             {
-                _oid = (value == null) ? null : new Oid(value);
+                _oid = (value is null) ? null : new Oid(value);
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedData.cs]: Please undo every `== null` to `is null` in this PR.

File: src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedData.cs [hunk 5]
```diff
@@ -13,31 +11,40 @@ public class AsnEncodedData
...
         {
+            _oid = null;
+            _rawData = Array.Empty<byte>();
         }
 
         public AsnEncodedData(byte[] rawData)
+            : this ((Oid?)null, rawData)
         {
-            Reset(null, rawData);
+            // nothing to do here
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedData.cs]: Why are you changing the flow to not call Reset?

**Comment by manne** [comment]
I am away from my dev environment for some days, and afterwards in Germany will be some public holidays. In ca. one week I will continue on this PR and incorporate the feedback. Thanks in advance

**Comment by stephentoub** [comment]
@manne, thank you for working on this.  It appears this was done in https://github.com/dotnet/runtime/pull/1535; @buyaa-n, why was that PR done when this PR was already in flight?  Regardless, since the work has been completed, I'm going to close this.  Thanks, again.

**Comment by manne** [comment]
@stephentoub I have learned something. That's awesome.

**Comment by buyaa-n** [comment]
Sorry @manne I didn't know there is this PR. In the future If you are working on any API for nullable annotation or raising PR please check corresponding check box here https://github.com/dotnet/corefx/issues/40623 or at least mention this issue in the PR so that we can keep track of nullability related issue/PRs. Thank you for your contribution

**Comment by manne** [comment]
@buyaa-n Thank you for this information. In the future I will "link" the PR with the issue you mentioned

**Pull request closed without merging**